src/commands/base-command.js

```diff
@@ -470,7 +470,10 @@ class BaseCommand extends Command {
       return await resolveConfig({
         config: options.config,
         cwd,
-        context: options.context || this.name(),
+        context:
+          options.context ||
+          process.env.CONTEXT ||
+          (['dev', 'dev:exec', 'dev:trace'].includes(this.name()) ? 'dev' : undefined),
```

Comment:
This was previously done in `oclif` child class, not sure if there's a better way to do it now.

Reply:
I don't think either. I would add a small comment here in the code to outline why we are checking for (`(['dev', 'dev:exec', 'dev:trace']` to help our future selfs when touching this code again 😂

Reply:
Good point, done in https://github.com/netlify/cli/pull/4040/commits/7ba63676d452b0a5fe8e4a1a6eaa5a2329ebdaf2